src/Shop.java

```diff
@@ -22,57 +22,114 @@ public Shop(String name, int floor, int box, List<String> types){
+            if (currentProduct.product.id == product.id) {
+                resultProduct = currentProduct;
+            }
+        }
+
+        if (resultProduct != null) {
+            resultProduct.quantity += quantity;
+        } else {
+            resultProduct = new ProductDetails(product, quantity);
+            products.add(resultProduct);
+        }
+    }
+
+    public Product getProduct(int id){
+        for (ProductDetails current : products) {
+            if (current.product.id == id) {
+                return current.product;
+            }
+        }
+        return null;
+    }
+
+    public ProductDetails findProductDetails(int id){
+        for(ProductDetails productDetails : products){
+            if(productDetails.product.id == id){
+                return productDetails;
+            }
+        }
+        return null;
+    }
+
+    public void updateProduct(int id, Product product){
+        ProductDetails toModify = findProductDetails(id);
+        deleteProduct(id);
+        product.id = id;
+        toModify.product = product;
+    }
+
+    public void deleteProduct(int id) {
         ProductDetails currentProduct = null;
-        for(int i = 0; i < products.size(); i++){
-            currentProduct = products.get(i);
-            if(currentProduct.product.name.equals(product.name)){
+        for (ProductDetails product : products) {
+            currentProduct = product;
+            if (currentProduct.product.id == id) {
                 break;
             }
         }
-
         if(currentProduct != null){
-            currentProduct.qualitity += qualitity;
-        } else {
-            currentProduct = new ProductDetails(product, qualitity);
-            products.add(currentProduct);
+            products.remove(currentProduct);
         }
     }
 
-    void hire(Employee employee){
+    public void hire(Employee employee) {
         employees.add(employee);
     }
 
-    void fire(Employee employee){
+    public void fire(Employee employee) {
         employees.remove(employee);
     }
 
-    boolean isEmployeeAvilable(Employee employee){
-        for(int i = 0; i < employees.size(); i++){
-            Employee currentEmployee = employees.get(i);
-            if(currentEmployee.equals(employee)){
-                if(currentEmployee.isAvailable) {
-                    return true;
-                } else {
-                    return false;
-                }
+    public void addService(Service service) {
+        services.add(service);
+    }
+
+    public Service getService(int id){
+        for(Service service : services){
+            if(service.id == id){
+                return service;
+            }
+        }
+        return null;
+    }
+
+    public void updateService(int id, Service modifiedService){
+        deleteService(id);
+        modifiedService.id = id;
+        addService(modifiedService);
+    }
+
+    public void deleteService(int id) {
+        for (Service service : services) {
```

Comment:
we should avoid modifying (remove/add) list during iteration.
try finding a service first (with `getService`) and then removing it